test/gator/test/fixtures/manifests/with-policies/with-violations-and-defaults.yaml

```diff
@@ -0,0 +1,75 @@
+apiVersion: templates.gatekeeper.sh/v1
+kind: ConstraintTemplate
+metadata:
+  name: k8srequiredlabelsdefault
+spec:
+  crd:
+    spec:
+      names:
+        kind: K8sRequiredLabelsDefault
+      validation:
+        legacySchema: false
+        openAPIV3Schema:
+          properties:
+            labels:
+              description: A list of labels and values the object must specify.
+              default:
+                - key: "aRequiredLabel"
+              items:
+                properties:
+                  key:
+                    description: The required label.
+                    type: string
+                type: object
+              type: array
+            message:
+              default: "aRequiredMessage"
+              type: string
+            placeholder: # needed to showcase defaulting of the other two props above.
+              type: string
+          type: object
+  targets:
+  - rego: |
+      package k8srequiredlabels
+
+      get_message(parameters, _default) = msg {
+        not parameters.message
+        msg := _default
+      }
+
+      get_message(parameters, _default) = msg {
+        msg := parameters.message
+      }
+
+      violation[{"msg": msg, "details": {"missing_labels": missing}}] {
+        provided := {label | input.review.object.metadata.labels[label]}
+        required := {label | label := input.parameters.labels[_].key}
+        missing := required - provided
+        count(missing) > 0
+
+        msg := input.parameters.message
+      }
+
+    target: admission.k8s.gatekeeper.sh
+---
+apiVersion: constraints.gatekeeper.sh/v1beta1
+kind: K8sRequiredLabelsDefault
+metadata:
+  name: all-must-have-owner
+spec:
+  match:
+    kinds:
+      - apiGroups: [""]
+        kinds: ["Namespace"]
+  parameters:
+    placeholder: "placeholder"
```

Comment:
So for https://github.com/open-policy-agent/gatekeeper/issues/2638, the user's example would work?

```yaml
properties:
  somekey:
    type: array
    default: ["somedefault"]
    items:
      type: string
```

Can you please share what the ConstraintTemplate would look like to support this?

Reply:
Yes, these changes are in support for what that issue describes and the snippet of open api v3 props should work as expected. I don't think the CT was shared in that issue.

For the tests I'm adding here, in particular, both the ConstraintTemplate and the Constraint are added.